.circleci/cimodel/data/caffe2_build_data.py

```diff
@@ -4,29 +4,29 @@
 
 CONFIG_TREE_DATA = [
     (Ver("ubuntu", "16.04"), [
-        (Ver("cuda", "9.0"), [
+        ([Ver("cuda", "9.0")], [
             # TODO make explicit that this is a "secret TensorRT build"
             #  (see https://github.com/pytorch/pytorch/pull/17323#discussion_r259446749)
             # TODO Uh oh, were we supposed to make this one important?!
             X("py2"),
             XImportant("cmake"),
         ]),
-        (Ver("cuda", "10.1"), [XImportant("py3.5")]),  # TensorRT 6 build
-        (Ver("mkl"), [XImportant("py2")]),
-        (Ver("gcc", "5"), [XImportant("onnx_py2")]),
-        (Ver("clang", "3.8"), [X("py2")]),
-        (Ver("clang", "3.9"), [X("py2")]),
-        (Ver("clang", "7"), [XImportant("py2"), XImportant("onnx_py3.6")]),
-        (Ver("android"), [XImportant("py2")]),
+        ([Ver("cuda", "10.1")], [XImportant("py3.5")]),  # TensorRT 6 build
+        ([Ver("mkl")], [XImportant("py2")]),
+        ([Ver("gcc", "5")], [XImportant("onnx_py2")]),
+        ([Ver("clang", "3.8")], [X("py2")]),
+        ([Ver("clang", "3.9")], [X("py2")]),
+        ([Ver("clang", "7")], [XImportant("py2"), XImportant("onnx_py3.6")]),
+        ([Ver("android")], [XImportant("py2")]),
     ]),
     (Ver("centos", "7"), [
-        (Ver("cuda", "9.0"), [X("py2")]),
+        ([Ver("devtoolset", "7"), Ver("cuda", "9.2")], [X("py3.6")]),
```

Comment:
Is this the only multi-element list?  Can you elaborate on this design choice?

Reply:
This enables us to have different compiler version controls for cpu files and gpu files. e.g. for this one we use gcc7 (comes from devtoolset7) for cpu files and nvcc 9.2 for gpu files.

Reply:
In the list representation, is it the case that at most there would only ever be 2 elements?  Seems that we should instead have separate, dedicated fields for the GPU compiler version and CPU compiler version.  If you don't want to do that now, add a detailed comment [in the class here](https://github.com/pytorch/pytorch/pull/28465/files#r338715224) that explains the situation.